Backpack/Skeleton/Classes/BPKSkeleton.swift

```diff
@@ -0,0 +1,166 @@
+        BPKColor.dynamicColor(withLightVariant: BPKColor.skyGrayTint06, darkVariant: BPKColor.blackTint02)
+    static internal var defaultColor: UIColor =
+        BPKColor.dynamicColor(withLightVariant: BPKColor.white, darkVariant: BPKColor.black)
+    private var customSize: CGSize = .zero
+    
+    public var type: BPKSkeletonType = .image {
+        didSet {
+            updateType()
+        }
+    }
+    
+    public var size: BPKSkeletonSize = .default {
+        didSet {
+            updateSize()
+        }
+    }
+    
+    public var style: BPKSkeletonStyle = .default {
+        didSet {
+            updateStyle()
+        }
+    }
+    
+    private var skeletonView: UIView = {
+        let view = UIView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.clipsToBounds = true
+        return view
+    }()
+    
+    public init(
+        type: BPKSkeletonType,
+        size: BPKSkeletonSize = .default,
+        style: BPKSkeletonStyle = .default
+    ) {
+        super.init(frame: .zero)
+        self.type = type
+        self.size = size
+        self.style = style
+        
+        setup()
+    }
+    
+    public required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        setup()
+    }
+    
+    private func updateType() {
+        switch self.type {
+        case .bodytext:
+            self.skeletonView = BPKTextSkeleton(size: viewSize)
+            self.skeletonView.backgroundColor = .clear
+        default:
+            self.skeletonView = BPKCommonSkeleton()
+        }
+    }
+    
+    private func setup() {
+        updateType()
+        addSubview(skeletonView)
+        updateSize()
+        updateStyle()
+    }
+    
+    private func updateStyle() {
+        if type == .image {
+            skeletonView.layer.cornerRadius = cornerStyle
+        }
+        if type == .headline {
+            skeletonView.layer.cornerRadius = BPKCornerRadiusXs
+        }
+        if type == .circle {
+            skeletonView.layer.cornerRadius = min(viewSize.height, viewSize.width) / 2.0
+        }
+    }
+    
+    private func updateSize() {
+        NSLayoutConstraint.activate([
+            skeletonView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            skeletonView.topAnchor.constraint(equalTo: topAnchor),
+            skeletonView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            skeletonView.bottomAnchor.constraint(equalTo: bottomAnchor),
+            
+            skeletonView.widthAnchor.constraint(equalToConstant: viewSize.width),
+            skeletonView.heightAnchor.constraint(equalToConstant: viewSize.height)
+        ])
+    }
+
+    public static func startShimmer(view: UIView) {
+        view.layer.sublayers?.filter { $0 is BPKShimmerLayer }.forEach { $0.removeFromSuperlayer() }
+        
+        let gradientLayer = BPKShimmerLayer()
+        let transperant = BPKSkeleton.defaultColor.withAlphaComponent(0).cgColor
```

Comment:
```suggestion
        let transparent = BPKColor.clear.cgColor
```
Tiny typo here

Reply:
Yea, the typo haha, my bad, but the color seems I cannot modify it. I got this with clear color:

https://user-images.githubusercontent.com/13347837/186820895-33426f54-3ecb-4a67-b1ba-0df798206021.mp4